pytorch_lightning/trainer/trainer.py

```diff
@@ -284,10 +287,21 @@ def __init__(
 
             benchmark: If true enables cudnn.benchmark.
 
+            deterministic: If true enables cudnn.deterministic
+
+            seed: integer number to ensure same results from different model runs
+
             terminate_on_nan: If set to True, will terminate training (by raising a `ValueError`) at the
                 end of each training batch, if any of the parameters or the loss are NaN or +/-inf.
         """
 
+        # seeding random number generators
+        if (seed is not None) and not deterministic:
+            log.warning("To ensure full reproducibility deterministic flag should also be set")
+        seed_everything(seed)
+        self.deterministic = deterministic
```

Comment:
I thought that this is the same, that is done by `benchmark` arg?

Reply:
Benchmark and deterministic is not the same, first one controls optimization of cudnn, second controls determinism(probably of kernel allocation or smth, can't really say). But it is two different parameters

Reply:
Topic on pytorch forum: https://discuss.pytorch.org/t/what-is-the-differenc-between-cudnn-deterministic-and-cudnn-benchmark/38054/3

Reply:
Or am I wrong?

Reply:
Sorry, that was my mistake. You're absolutely right.